client/src/graphics/animated_sprite.lua

```diff
@@ -10,25 +10,24 @@ local floor = math.floor
 
 Animation =
 class(
-  function (self, image, width, height, dur)
+  function (self, image, width, height)
     self.image = image
     self.frameSize = {width = width, height = height}
-    self.spriteSheet = love.graphics.newSpriteBatch(image)
     self.frames = {}
-    self.frameDuration = dur or 2
     self.loop = false
     self.loopStartFrame = 1
   end
 )
 
-function Animation.addFrame(self, frame, length)
+function Animation.addFrame(self, frame, duration)
   local imgW, imgH = self.image:getDimensions()
   local w = self.frameSize.width
   local h = self.frameSize.height
   local x = w*(frame-1)%imgW
   local y = h*floor(w*(frame-1)/imgW)
-  tableInsert(self.frames, {quad(x, y, w, h, imgW, imgH), length})
+  self.frames[#self.frames+1] = { quad = quad(x, y, w, h, imgW, imgH), duration = duration }
```

Comment:
Working with `table[#table+1] = {}` is generally faster than table.insert.
Mostly commenting on this because you created a local to make the lookup for table.insert slightly faster.
Given this is code only running during the loading process and we don't need it to be as fast as possible using `table.insert` is perfectly fine but creating a local for it doesn't make sense in that situation.
Additionally I also replaced the reference to the quad with named references rather than indexes because readability is generally more important. I adjusted the references elsewhere accordingly.